blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/RunContainerImpl.java

```diff
@@ -59,8 +62,22 @@ public BlueRun get(String name) {
                     break;
                 }
             }
+
+            int number;
+            try {
+                number = Integer.parseInt(name);
+            } catch (NumberFormatException e) {
+                throw new NotFoundException(String.format("Run %s not found in organization %s and pipeline %s",
+                    name, pipeline.getOrganization(), job.getName()));
+            }
+            for (BlueQueueItem item : QueueContainerImpl.getQueuedItems(job)) {
```

Comment:
the client does not handle this today and that comes at a high price of complexity there. It depends where we want to wear the ugliness.